alma/controllers/hook/displayPayment.php

```diff
@@ -36,77 +36,49 @@ class AlmaDisplayPaymentController extends AlmaProtectedHookController
 {
     public function run($params)
     {
-        // First check that we can offer Alma for this payment
-        $eligibility = AlmaEligibilityHelper::eligibilityCheck($this->context);
-
-        $error = false;
-        if (!$eligibility) {
-            $error = true;
-        }
-
-        if (isset($eligibility) && $eligibility->isEligible) {
-            $disabled = false;
-        } else {
-            if (AlmaSettings::showDisabledButton()) {
-                $disabled = true;
-            } else {
-                return null;
-            }
-        }
 
         if (is_callable('Media::getMediaPath')) {
             $logo = Media::getMediaPath(_PS_MODULE_DIR_ . $this->module->name . '/views/img/alma_payment_logos.svg');
         } else {
             $logo = $this->module->getPathUri() . '/views/img/alma_payment_logos.svg';
         }
 
-        $cart = $params['cart'];
-        $purchaseAmount = almaPriceToCents((float) $cart->getordertotal(true, Cart::BOTH));
+        $installmentPlans = AlmaEligibilityHelper::eligibilityCheck($this->context);
```

Comment:
Do something when `$installmentPlans` is empty:

- Easy option: return
- Or: if disabled button display is enabled, loop over merchant-activated installments plan and show a disabled button for each (a payment option each with `$plans => null, $disabled => true, $error > true`)